### ISSUE TITLE
Simple count mv sanity check failure

### ISSUE THREAD

**Issue by chenzl25** [risingwavelabs/risingwave#6343]
### Describe the bug

```
dev=> explain create materialized view v as select count(*) c from sbtest1;
                           QUERY PLAN
----------------------------------------------------------------
 StreamMaterialize { columns: [c], pk_columns: [] }
 └─StreamProject { exprs: [sum0(count)] }
   └─StreamGlobalSimpleAgg { aggs: [sum0(count), sum0(count)] }
     └─StreamExchange { dist: Single }
       └─StreamStatelessLocalSimpleAgg { aggs: [count, count] }
         └─StreamTableScan { table: sbtest1, columns: [id] }
```

```
thread 'risingwave-streaming-actor' panicked at 'inconsistent update!
table_id: 1005, vnode: 0, key: Row([])
stored value: Some(Row([Some(Int64(80780)), Some(Int64(80780))]))
expected value: Row([Some(Int64(205200)), Some(Int64(205200))])
new value: Row([Some(Int64(206224)), Some(Int64(206224))])', /Users/dylan/Desktop/playground/risingwave/src/storage/src/table/streaming_table/state_table.rs:735:13
stack backtrace:
   0: rust_begin_unwind
             at /rustc/b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at /rustc/b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1/library/core/src/panicking.rs:65:14
   2: risingwave_storage::table::streaming_table::state_table::StateTable<S>::do_update_sanity_check::{{closure}}
             at ./src/storage/src/table/streaming_table/state_table.rs:735:13
   3: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1/library/core/src/future/mod.rs:91:19
   4: risingwave_storage::table::streaming_table::state_table::StateTable<S>::batch_write_rows::{{closure}}
             at ./src/storage/src/table/streaming_table/state_table.rs:640:29
   5: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1/library/core/src/future/mod.rs:91:19
   6: risingwave_storage::table::streaming_table::state_table::StateTable<S>::commit::{{closure}}
             at ./src/storage/src/table/streaming_table/state_table.rs:589:57
   7: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1/library/core/src/future/mod.rs:91:19
   8: risingwave_stream::executor::global_simple_agg::GlobalSimpleAggExecutor<S>::flush_data::{{closure}}
             at ./src/stream/src/executor/global_simple_agg.rs:251:39
   9: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1/library/core/src/future/mod.rs:91:19
  10: risingwave_stream::executor::global_simple_agg::GlobalSimpleAggExecutor<S>::execute_inner::{{closure}}
             at ./src/stream/src/executor/global_simple_agg.rs:332:22
  11: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  12: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  13: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  14: futures_util::stream::stream::StreamExt::poll_next_unpin
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/mod.rs:1626:9
  15: <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/next.rs:32:9
  16: <minitrace::future::InSpan<T> as core::future::future::Future>::poll
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/minitrace-0.4.0/src/future.rs:118:19
  17: risingwave_stream::executor::wrapper::trace::trace::{{closure}}
             at ./src/stream/src/executor/wrapper/trace.rs:53:60
  18: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  19: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  20: futures_util::stream::stream::StreamExt::poll_next_unpin
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/mod.rs:1626:9
  21: <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/next.rs:32:9
  22: <async_stack_trace::StackTraced<F,_> as core::future::future::Future>::poll
             at ./src/utils/async_stack_trace/src/lib.rs:178:23
  23: risingwave_stream::executor::wrapper::trace::stack_trace::{{closure}}
             at ./src/stream/src/executor/wrapper/trace.rs:120:70
  24: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  25: risingwave_stream::executor::wrapper::schema_check::schema_check::{{closure}}
             at ./src/stream/src/executor/wrapper/schema_check.rs:24:1
  26: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  27: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  28: futures_util::stream::stream::StreamExt::poll_next_unpin
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/mod.rs:1626:9
  29: <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/next.rs:32:9
  30: risingwave_stream::executor::wrapper::epoch_check::epoch_check::{{closure}}
             at ./src/stream/src/executor/wrapper/epoch_check.rs:31:44
  31: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  32: risingwave_stream::executor::wrapper::update_check::update_check::{{closure}}
             at ./src/stream/src/executor/wrapper/update_check.rs:27:1
  33: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  34: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  35: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  36: risingwave_stream::executor::simple::SimpleExecutorWrapper<E>::execute_inner::{{closure}}
             at ./src/stream/src/executor/simple.rs:68:5
  37: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  38: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  39: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  40: futures_util::stream::stream::StreamExt::poll_next_unpin
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/mod.rs:1626:9
  41: <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/next.rs:32:9
  42: <minitrace::future::InSpan<T> as core::future::future::Future>::poll
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/minitrace-0.4.0/src/future.rs:118:19
  43: risingwave_stream::executor::wrapper::trace::trace::{{closure}}
             at ./src/stream/src/executor/wrapper/trace.rs:53:60
  44: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  45: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  46: futures_util::stream::stream::StreamExt::poll_next_unpin
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/mod.rs:1626:9
  47: <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/next.rs:32:9
  48: <async_stack_trace::StackTraced<F,_> as core::future::future::Future>::poll
             at ./src/utils/async_stack_trace/src/lib.rs:178:23
  49: risingwave_stream::executor::wrapper::trace::stack_trace::{{closure}}
             at ./src/stream/src/executor/wrapper/trace.rs:120:70
  50: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  51: risingwave_stream::executor::wrapper::schema_check::schema_check::{{closure}}
             at ./src/stream/src/executor/wrapper/schema_check.rs:24:1
  52: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  53: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  54: futures_util::stream::stream::StreamExt::poll_next_unpin
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/mod.rs:1626:9
  55: <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/next.rs:32:9
  56: risingwave_stream::executor::wrapper::epoch_check::epoch_check::{{closure}}
             at ./src/stream/src/executor/wrapper/epoch_check.rs:31:44
  57: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  58: risingwave_stream::executor::wrapper::update_check::update_check::{{closure}}
             at ./src/stream/src/executor/wrapper/update_check.rs:27:1
  59: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  60: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  61: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  62: risingwave_stream::executor::mview::materialize::MaterializeExecutor<S>::execute_inner::{{closure}}
             at ./src/stream/src/executor/mview/materialize.rs:125:5
  63: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  64: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  65: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  66: futures_util::stream::stream::StreamExt::poll_next_unpin
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/mod.rs:1626:9
  67: <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/next.rs:32:9
  68: <minitrace::future::InSpan<T> as core::future::future::Future>::poll
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/minitrace-0.4.0/src/future.rs:118:19
  69: risingwave_stream::executor::wrapper::trace::trace::{{closure}}
             at ./src/stream/src/executor/wrapper/trace.rs:53:60
  70: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  71: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  72: futures_util::stream::stream::StreamExt::poll_next_unpin
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/mod.rs:1626:9
  73: <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/next.rs:32:9
  74: <async_stack_trace::StackTraced<F,_> as core::future::future::Future>::poll
             at ./src/utils/async_stack_trace/src/lib.rs:178:23
  75: risingwave_stream::executor::wrapper::trace::stack_trace::{{closure}}
             at ./src/stream/src/executor/wrapper/trace.rs:120:70
  76: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  77: risingwave_stream::executor::wrapper::schema_check::schema_check::{{closure}}
             at ./src/stream/src/executor/wrapper/schema_check.rs:24:1
  78: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  79: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  80: futures_util::stream::stream::StreamExt::poll_next_unpin
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/mod.rs:1626:9
  81: <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.24/src/stream/stream/next.rs:32:9
  82: risingwave_stream::executor::wrapper::epoch_check::epoch_check::{{closure}}
             at ./src/stream/src/executor/wrapper/epoch_check.rs:31:44
  83: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  84: risingwave_stream::executor::wrapper::update_check::update_check::{{closure}}
             at ./src/stream/src/executor/wrapper/update_check.rs:27:1
  85: <futures_async_stream::try_stream::from_generator::GenTryStream<G> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-async-stream-0.2.5/src/lib.rs:430:31
  86: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  87: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.24/src/stream.rs:120:9
  88: <risingwave_stream::executor::dispatch::DispatchExecutor as risingwave_stream::executor::StreamConsumer>::execute::{{closure}}
             at ./src/stream/src/executor/dispatch.rs:260:9
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


*** async stack trace context of current task ***

Actor 25: `SELECT count(*) AS c FROM sbtest1` [13.335491291s]
  Epoch 3353341433413632 [447.514875ms]
    MaterializeExecutor 1900000043 (actor 25, executor 67) [447.505375ms]
      ProjectExecutor 1900000042 (actor 25, executor 66) [447.517749ms]
        GlobalSimpleAggExecutor-1900000040 (actor 25, executor 64) [447.5155ms]  <== current
```

### To Reproduce

1. Change `risingwave.toml`
```
barrier_interval_ms = 10
checkpoint_frequency = 2
```

2. Use sysbench to keep inserting data to table `sbtest1`
```
sysbench \
--db-driver=pgsql \
--pgsql-host=127.0.0.1  \
--pgsql-port=4566  \
--pgsql-user=root \
--pgsql-db=dev \
--threads=16 \
--report-interval=1 \
--events=0 \
--time=10 \
--percentile=99 \
--auto_inc=false \
--table_size=100000000 \
--db-ps-mode=disable \
--skip_trx=true \
--range_size=1 \
--random_points=1 \
/usr/local/share/sysbench/select_random_points.lua \
prepare
```

3. Create a mv (with another terminal so that we can run create mv concurrently with the insert statements generated by the sysbench)
```sql
 create materialized view v as select count(*) c from sbtest1;
```

4. Wait a minute and then it will fail.

**Comment by hzxa21** [comment]
Findings:
- There is indeed a data loss in reading the simple agg state via `get`
- Right before the data loss happens, there is a L0 -> L0 sub-level compaction triggered

Chatted with @Little-Wallace offline. We suspect this is due to the recent merged delete range support breaking some previous assumptions on SSTs.

**Comment by chenzl25** [comment]
Yes, I believe so. I have logged all the update of `GlobalSimpleAggExecutor` and find that at some point the executor read the state with latest epoch but read some old data far from now.

**Comment by hzxa21** [comment]
@Little-Wallace found out that this bug is introduced by #5998. Previously we assume for a given user key cannot exists in more than one SSTs in a given level and we use user key to do binary search during `get`.

After #5998, compactor uses full key instead of user key to find out overlapping SSTs to construct a compaction task to take delete range tombstone into consideration. However, this will break the assumption mentioned above and thus causing incorrect result in `get`. We plan to revert this change first in #6364 to fix the issue since range deletion is not used and refactoring on the range deletion side will be done in a separate PR.
